tools/deviceGen/output/traits_defaults.hlsl

```diff
@@ -0,0 +1,382 @@
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating16BitUnsignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating16BitSignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating16BitMixedSignednessAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating32BitUnsignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating32BitSignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating32BitMixedSignednessAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating64BitUnsignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating64BitSignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating64BitMixedSignednessAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(storageTexelBufferOffsetAlignmentBytes, size_t, std::numeric_limits<size_t>::max());
+NBL_GENERATE_GET_OR_DEFAULT(uniformTexelBufferOffsetAlignmentBytes, size_t, std::numeric_limits<size_t>::max());
+NBL_GENERATE_GET_OR_DEFAULT(maxBufferSize, size_t, MinMaxSSBOSize);
+// Nabla Core Extensions
+NBL_GENERATE_GET_OR_DEFAULT(minImportedHostPointerAlignment, uint32_t, 2147483648);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat32AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat64AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat32AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat64AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderImageFloat32AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(sparseImageFloat32Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(sparseImageFloat32AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(robustStorageBufferAccessSizeAlignment, size_t, 9223372036854775808);
+NBL_GENERATE_GET_OR_DEFAULT(robustUniformBufferAccessSizeAlignment, size_t, 9223372036854775808);
+// Extensions
+NBL_GENERATE_GET_OR_DEFAULT(shaderTrinaryMinmax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderExplicitVertexParameter, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(gpuShaderHalfFloatAMD, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderImageLoadStoreLod, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(displayTiming, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(maxDiscardRectangles, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(primitiveOverestimationSize, float, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxExtraPrimitiveOverestimationSize, float, 0);
+NBL_GENERATE_GET_OR_DEFAULT(extraPrimitiveOverestimationSizeGranularity, float, std::numeric_limits<float>::infinity());
```

Comment:
there's `nbl::hlsl::numeric_limits<float>::infinity` its usable both from C++ and HLSL, and provided by `limits.hlsl` IIRC

Reply:
why did this inf turn into a 0 on your latest codegen?